app/code/Magento/InventorySales/Model/ReturnProcessor/GetShippedItemsPerSourceByPriority.php

```diff
@@ -8,16 +8,19 @@
 namespace Magento\InventorySales\Model\ReturnProcessor;
 
 use Magento\Framework\Exception\InputException;
-use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Model\Order as OrderModel;
 use Magento\Sales\Model\Order\Shipment;
 use Magento\Sales\Api\Data\OrderItemInterface;
 use Magento\InventoryApi\Api\GetSourcesAssignedToStockOrderedByPriorityInterface;
 use Magento\InventoryCatalogApi\Model\GetSkusByProductIdsInterface;
 use Magento\InventorySalesApi\Model\StockByWebsiteIdResolverInterface;
 use Magento\InventoryShipping\Model\ResourceModel\ShipmentSource\GetSourceCodeByShipmentId;
 use Magento\Framework\Exception\LocalizedException;
+use Magento\InventorySales\Model\ReturnProcessor\Result\SourceDeductedOrderItemInterfaceFactory;
+use Magento\InventorySales\Model\ReturnProcessor\Result\SourceDeductedOrderItemsResultInterfaceFactory;
+use Magento\InventorySales\Model\ReturnProcessor\Result\SourceDeductedOrderItemsResultInterface;
 
-class GetShippedItemsPerSourceByPriority
+class GetShippedItemsPerSourceByPriority implements GetSourceDeductedOrderItemsInterface
```

Comment:
This class should be moved to InventoryShipping module, along with that we would fix circular dependency InventoryShipping -> InventorySales and vice versa.

fixing that we would have 
InventoryShipping -> InventorySales
and InventorySales does not depend on InventoryShipping